src/layouts/ErrorBoundary.js

```diff
@@ -6,32 +6,23 @@ import { logError } from '../helpers'
 export default class ErrorBoundary extends Component {
   constructor(props) {
     super(props)
-    this.state = {
-      error: '',
-      errorInfo: '',
-      hasError: false,
-    }
-  }
-
-  static getDerivedStateFromError(_error) {
-    if (_error) {
-      return { hasError: true }
-    }
+    this.state = { error: null, errorInfo: null }
   }
 
   componentDidCatch(error, errorInfo) {
-    this.setState({ errorInfo, error })
+    this.setState({
+      error: error,
+      errorInfo: errorInfo,
+    })
```

Comment:
I'm confused because i don't find that setState method was declared. At this case don't should u to use `this.state` directly?

Reply:
setState method is inherited from parent at the line 8?

Reply:
I checked and is good. This is the documentation that I check:

https://es.reactjs.org/docs/composition-vs-inheritance.html